### ISSUE TITLE
Naval Warfare: pressing the navigation keys and set a destination shall override each other

### ISSUE THREAD

**Issue by hessenfarmer** [widelands/widelands#5981]
**Describe the bug**
When a destination is set for a ship, pressing the navigation keys does not have an effect.

**To reproduce**
1. Set a  war or expedition ship to go to a destination
2. when ship is underway try to use the navigation buttons to alter the direction of a ship
3. Navigation keys do not ahave an effect unless destination is manually set to none.

**Expected behavior**
Each alteration in the navigation shall override last settings. pressing a key cancels destination. setting destination overrides directional commands.

**Comment by Noordfrees** [comment]
Cannot reproduce, the direction buttons always clear the destination and take effect immediately

**Comment by Noordfrees** [comment]
Ah, now I see what the problem is: The destination buttons still *work*, but while the ship is sailing to a destination the status of swimmable and explorable directions is not updated, so the buttons remain the same as they were the last time the ship didn't have a destination.
